### ISSUE TITLE
`setRowData` will rerender the entire grid, no matter what was changed. There is no vitrtual DOM diffing happening. Causing huge performance issues.

### ISSUE THREAD

**Issue by eelyafi** [ceolter/ag-grid-react#66]
**I'm submitting a ...**  (check one with "x")
```
[x] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/ceolter/ag-grid-react/blob/master/CONTRIBUTING.md#question
```

**Current behavior**
`setRowData` will rerender the entire grid, no matter what was changed. There is no vitrtual DOM diffing happening. Causing huge performance issues.

**Expected behavior**
`setRowData` will rerender only changed cells. 

**Minimal reproduction of the problem with instructions**
call `setRowData` whenever data is changed.

**What is the motivation / use case for changing the behavior?**
Huge performance impact

**Please tell us about your environment:**
MacOS, Chrome, React + webpack

* **ag-grid-react version:** 8.2.0

* **ag-Grid/ag-Grid-Enterprise version:** 2.0.X

* **React version:** 15.4.1

* **Browser:** [all | Chrome XX | Firefox XX | IE XX | Safari XX | Mobile Chrome XX | Android X.X Web Browser | iOS XX Safari | iOS XX UIWebView | iOS XX WKWebView ]
ALL

* **Language:** [all | TypeScript X.X | ES6/7 | ES5]
ES6

**Comment by seanlandsman** [comment]
setRowData is a blanket data replacement - it isnt intended for diffing/merging of row data

If you want to merge/diff only changed rows then you should use the corresponding API calls available to you. 

See here for more information around this:

https://www.ag-grid.com/javascript-grid-insert-remove/?framework=react#gsc.tab=0
https://www.ag-grid.com/javascript-grid-refresh/?framework=react#gsc.tab=0

**Comment by eelyafi** [comment]
@seanlandsman what was the reason of not using the power of React ?

**Comment by eelyafi** [comment]
@seanlandsman I used `refreshRows` in order to optimize grid. Now, when `breakdown` property is being changed the rows are not updated..

**Comment by uvtzxpm** [comment]
ag-grid is not really written with React in mind, it also supports being used with no framework, and many other frameworks. ag-grid-react is only a very thin wrapper over ag-grid. If you want a grid which uses React all the way down, you'll have to look elsewhere - but most of those grids are missing various features that ag-grid has.

ag-grid does already virtualise rows/columns - it doesn't keep everything in the DOM at the same time. Take a look at the `rowBuffer` and `suppressColumnVirtualisation` properties:
https://www.ag-grid.com/javascript-grid-properties/#gsc.tab=0

Basically, your bug report is not very useful without a live test case. If you can demonstrate a performance issue, maybe the ag-grid team can help you out a bit.
